opty/direct_collocation.py

```diff
@@ -438,6 +440,13 @@ def plot_constraint_violations(self, vector, axes=None):
         axes : ndarray of AxesSubplot
             A matplotlib axes with the constraint violations plotted.
 
+            If axes are given by the user:
+            len(axes) = 2 if len(self.collocator.instance_constraints) <= bars_per_plot.
+            len(axes) = len(self.collocator.instance_constraints) // bars_per_plot + 2
+                        if len(self.collocator.instance_constraints) % bars_per_plot != 0.
+            len(axes) = len(self.collocator.instance_constraints) // bars_per_plot + 1
+                        if len(self.collocator.instance_constraints) % bars_per_plot = 0.
```

Comment:
> There is no way for the user to know what the value of `bars_per_plot` is in the docstring. Just state the number. This could be more simply stated in general. Something like, "six instance constraints will be plotted per axes, so you'll need to pass in enough rows to accomodate the number of instance constraints".

My approach was this: I select between 6 to 10 bars per plot to prevent situations like with the double pendulum., where there is only on bar on the last plot. If the wrong number of axes is given, the method will tell the user how many are needed - unless he only gives one, which is always wrong.